syncer/optimist.go

```diff
@@ -91,6 +91,12 @@ func (s *Syncer) handleQueryEventOptimistic(
 		err      error
 	)
 
+	err = s.execError.Get()
+	if err != nil {
+		ec.tctx.L().Error("error detected when executing SQL job", log.ShortError(err))
+		return nil
```

Comment:
if we didn't return error, syncer's loop will continue to get the next event. Not sure things will more complex 🤔

Reply:
`sync` and `syncDDL` will cancel the loop. Same for [other mode](https://github.com/pingcap/dm/blob/ef6138bf373249f7f9afa8a6d112f555f59e5bc4/syncer/syncer.go#L1940).